### ISSUE TITLE
sum(sparse) -> dense?

### ISSUE THREAD

**Issue by yurivish** [JuliaSparse/SparseArrays.jl#43]
```julia
julia> A = sparse([rand() < 0.01 ? 1 : 0 for _ in 1:50, _ in 1:50])
50×50 SparseMatrixCSC{Int64,Int64} with 23 stored entries:
[...]

julia> sum(A, dims=1)
1×50 Array{Int64,2}:
 0  1  0  0  0  0  0  0  1  0  1  0  0  0  0  1  0  1  0  0  1  1  0  0  0  2  3  0  0  0  0  0  0  0  0  0  0  0  1  1  1  0  0  1  1  0  2  1  0  3
```

Sometimes you have a very sparse matrix and want to sum a slice of it. The slice may potentially have very many columns or rows that are entirely zero, in which case it makes a lot of sense to preserve sparsity in the output.

In the non-slice case, where every row and column usually contains at least one nonzero value, it makes sense to keep the result dense.

It would be great if preserving sparsity in `sum` and other similar reductions were allowed via a keyword argument, eg. `sparse=true`.

cc @simonbyrne, who suggested the keyword idea on Slack a few months ago.

**Comment by SobhanMP** [comment]
@Wimmerer do you think this would be a good feature to add?

**Comment by ViralBShah** [comment]
Yes, this is good to do - which is make the sum of SparseMatrixCSC along either dimension a SparseVector.

**Comment by Wimmerer** [comment]
I imagine this SparseVector will *very* often be dense but that's the only correct route.
